src/network/networkd-network.c

```diff
@@ -167,6 +167,10 @@ static int network_load_one(Manager *manager, const char *filename) {
         if (network->ip_masquerade)
                 network->ip_forward |= ADDRESS_FAMILY_IPV4;
 
+        /* IPv6ProxyNDPAddress entries imply IPv6ProxyNDP=yes */
+        if (network->n_ipv6_proxy_ndp_addresses > 0)
+                network->ipv6_proxy_ndp = 1;
```

Comment:
Hm, I think this should be conditionalized on `network->ipv6_proxy_ndp < 0`. If `IPv6ProxyNDP=no` was explicitly set, it shouldn't be overridden.